lib/ansible/utils/display.py

```diff
@@ -161,19 +170,24 @@ def display(self, msg, color=None, stderr=False, screen_only=False, log_only=Fal
                     raise
 
         if logger and not screen_only:
-            msg2 = nocolor.lstrip(u'\n')
+            # We first convert to a byte string so that we get rid of
+            # color and characters that are invalid in the user's locale
+            msg2 = to_bytes(nocolor.lstrip(u'\n'))
 
-            msg2 = to_bytes(msg2)
             if sys.version_info >= (3,):
                 # Convert back to text string on python3
-                # We first convert to a byte string so that we get rid of
-                # characters that are invalid in the user's locale
                 msg2 = to_text(msg2, self._output_encoding(stderr=stderr))
 
-            if color == C.COLOR_ERROR:
-                logger.error(msg2)
-            else:
-                logger.info(msg2)
+            lvl = logging.INFO
+            if color:
+                # set logger level based on color (not great)
+                try:
+                    lvl = color_to_log_level[color]
+                except KeyError:
+                    # this should not happen, but JIC
```

Comment:
I wonder if we should `raise AnsibleAssertionError` here then?

Reply:
Until now log/callback errors have been 'non fatal' , that would be a change in that behaviour

Reply:
Right, though passing a non-existing color into `display()` would be a programming error, no?